File: java/side exercises/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.List;
import java.util.ArrayList;

class ResistorColorDuo {
	
	List<String> resistors = new ArrayList<>();
	
	ResistorColorDuo () {
		resistors.add ("black");
		resistors.add ("brown");
		resistors.add ("red");
		resistors.add ("orange");
		resistors.add ("yellow");
		resistors.add ("green");
		resistors.add ("blue");
		resistors.add ("violet");
		resistors.add ("grey");
		resistors.add ("white");
	}
	
    int value(String[] colors) {
        int code = resistors.indexOf(colors[0]);
		
		return colors.length > 1 ? code * 10 + resistors.indexOf(colors[1]) : code;
    }
}

File: csharp/side exercises/resistor-color-duo/ResistorColorDuo.cs
using System;
using System.Collections.Generic;

public static class ResistorColorDuo
{
    private static readonly List<string> resistorColors = new List<string>()
    {
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white"
    };

    public static int Value(string[] colors) => resistorColors.IndexOf(colors[0]) * 10 + 
                                                resistorColors.IndexOf(colors[1]);

}
